core/geometry/src/polyface/IndexedPolyfaceWalker.ts

```diff
@@ -0,0 +1,350 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module Polyface
+ */
+
+import { IndexedEdgeMatcher, SortableEdgeCluster } from "./IndexedEdgeMatcher";
+import { IndexedPolyfaceVisitor } from "./IndexedPolyfaceVisitor";
+import { IndexedPolyface } from "./Polyface";
+
+/**
+ * The `IndexedPolyfaceWalker` class supports navigation around facets, across edges, and around vertices in an
+ * [[IndexedPolyface]].
+ * * Compare to the [[IndexedPolyfaceVisitor]] class, which supports the iteration of facets in an `IndexedPolyface`.
+ * * A one-time call to [[buildEdgeMateIndices]] creates and populates the `data.edgeMateIndex` array on the input.
+ *   * This array essentially completes the topology of the `IndexedPolyface` by storing facet adjacency.
+ * * After this setup, caller code can create `IndexedPolyfaceWalker` objects via:
+ *   * `walker = IndexedPolyfaceWalker.createAtFacetIndex(polyface, facetIndex, offsetWithinFacet)`
+ *   * `walker = IndexedPolyfaceWalker.createAtEdgeIndex(polyface, edgeIndex)`
+ *   * `walker = IndexedPolyfaceWalker.createAtVisitor(visitor, offsetWithinFacet)`
+ * * Once you have a walker object, you can traverse the face, edge, and vertex loops it references. For
+ * example, if `walker.edgeIndex === A`, referring to the right edge of the upper left facet pictured below, then
+ * the following are true:
+ *   * `walker.nextAroundFacet().edgeIndex === B`
+ *   * `walker.previousAroundFacet().edgeIndex === C`
+ *   * `walker.edgeMate().edgeIndex === F`
+ *   * `walker.nextAroundVertex().edgeIndex === E`
+ *   * `walker.previousAroundVertex().edgeIndex === D`
+ * * When facets are viewed so that the face loops stored in the [[PolyfaceData]] `pointIndex` array have
```

Comment:
I move the image (line 59-70) to line 32

Reply:
To be addressed in next commit.